### PR TITLE
[RFR] Categories

### PR DESCRIPTION
**Pull request by mchaffotte**
- Add a new filter to get products by categories
- Add a page to show all categories with a link to product page

### CODE REVIEW

File: cypress/integration/app.spec.js [hunk 1]
```diff
@@ -48,4 +48,13 @@ describe('Global app Tests', () => {
         cy.get('button[type="submit"]').click();
         cy.contains('Element updated').should('be.visible');
     });
+
+    it('Filter products using category page', () => {
```
Comment (floo51) [inline cypress/integration/app.spec.js]: it filter**s**

or 

it should filter

File: cypress/integration/app.spec.js [hunk 2]
```diff
@@ -48,4 +48,13 @@ describe('Global app Tests', () => {
...
+        cy.visit('/');
+        cy.contains('categories').click();
+        cy.get('a[role="button"]')
```
Comment (floo51) [inline cypress/integration/app.spec.js]: This gonna break fast.

Avoid selectors based on dom structure. Base them on displayed text if possible.

File: src/customer/segments.js [hunk 3]
```diff
@@ -38,7 +38,7 @@ export const SegmentList = () => (
                         <TableCell>{segment}</TableCell>
                         <TableCell>
                             <Button
-                                href={`#/customers?filter=%7B"groups"%3A%5B"${segment}"}"%5D%7D`}
+                                href={`#/customers?filter=%7B"groups"%3A%5B"${segment}"%5D%7D&page=1&perPage=25`}
```
Comment (floo51) [inline src/customer/segments.js]: Not sure why you force pagination here? Let the user decide on their own.

File: src/product/categories.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+        <Datagrid>
+            <TextField source="name" />
+            <ProductLink />
+        </Datagrid>
+    </List>
+);
+
+const ProductLink = ({ record }) => (
+    <Button
+        href={`#/products?filter=%7B"category_id"%3A${record.id}%7D&order=DESC&page=1&perPage=25&sort=id}`}
```
Comment (floo51) [inline src/product/categories.js]: Same here + the sort

File: src/product/products.js [hunk 5]
```diff
@@ -67,13 +70,24 @@ export const ProductEdit = props => (
     </Edit>
 );
 
+const ProductFilter = props => (
```
Comment (floo51) [inline src/product/products.js]: `ProductFilters` would be a more accurate naming IMO.

Right now, it feels like you filter something by products, not product by something else

File: src/customer/segments.js (hunk 3 above)
Comment (mchaffotte) [reply src/customer/segments.js]: it is mandatory to filter with a redirection. It should be better to get the pagination preferences from redux. but we do not do that for the moment.

File: cypress/integration/app.spec.js [hunk 6]
```diff
@@ -32,20 +38,45 @@ describe('Global app Tests', () => {
...
+            .find('input')
+            .type('10');
+        cy.contains('div', 'Price')
+            .find('input')
+            .type('10000');
+        cy.contains('div', 'Stock')
+            .find('input')
+            .type('1');
+        cy.contains('Save').click();
+        cy.contains('Save').click();
```
Comment (floo51) [inline cypress/integration/app.spec.js]: There probably is an issue here?

**Pull request merged**